### PR TITLE
Update modules docs and docs generating system

### PR DESCRIPTION
**Pull request by kuraga**
Use doc files from `node_modules` directory but from `HEAD` branch of github repo.

### CODE REVIEW

File: bin/modules-docs.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var path = require('path');
+var fs = require('fs');
+var https = require('https');
+var logger = require('console');
+
+var docs = path.join(__dirname, '..', 'docs', 'modules');
+var moduleReadmes = {
+    'geval': ['Raynos/geval', '2.1.1', 'README.md'],
```
Comment (Raynos) [inline bin/modules-docs.js]: hard coding the version number is going to be frustrating. I recommend you read it from the README and ideally query npm for the latest version matching the semver range.

Alternatively just fetch the HEAD from the github repository.

File: bin/modules-docs.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        'at version ' + moduleData[1] + '.\n\n', function writeModuleDoc() {
+            writeModuleDocFile(file, moduleName, moduleReadmeUri, moduleReadmeFile);
+        });
+}
+
+function writeModuleDocFile(file, moduleName, moduleReadmeUri, moduleReadmeFile) {
+    https.get(moduleReadmeUri, function onModuleReadmeFileGot(response) {
+        response.pipe(file);
+        file.on('finish', function onModuleReadmeFileSaved() {
+            file.close(function onWriteModuleDocSuccess() {
```
Comment (Raynos) [inline bin/modules-docs.js]: you do not have the close files. They should close on `end()`.

File: bin/modules-docs.js (hunk 1 above)
Comment (kuraga) [inline bin/modules-docs.js]: The idea is opposite: they *should be equal* to versions in `package.json`. Agree? 

Ho to do it?

Comment (kumavis) [inline bin/modules-docs.js]: If you store the version numbers here you'll have to manually or programmatically update this file. both sound like a bad idea. Instead grab the version from the package.json. Also, why are we downloading from github when they should already be in our node_modules folder?

Comment (kuraga) [inline bin/modules-docs.js]: Yeah, nice idea. Or `npm.getCurrentPackageVersion('geval')`, or grab file from `node_modules`. I prefer the second. I'll make.

Comment (Raynos) [inline bin/modules-docs.js]: grabbing from node_modules is fine with me.

**Comment by Raynos** [comment]
suggestion: To make reviews easier you can consider doing multiple commits. For example one commit where you change binmodules-docs.js module and a second commit where you run it and change the auto generated docs files.

**Comment by kuraga** [comment]
Ok

**Comment by kuraga** [comment]
Updated. But there is a problem: we can't get version of packages are used in docs.

**Comment by Raynos** [comment]
@kuraga we sure can.

You can read both the README.md and package.json out of node_modules. The package.json file will contain the version.

**Comment by kuraga** [comment]
Hm... It's just JSON, yeah!!!

**Comment by kuraga** [comment]
@Raynos updated!

**Comment by Raynos** [comment]
lgtm.

**Comment by kuraga** [comment]
Thanks!

**Pull request merged**